workers/unity/Assets/Fps/Scripts/GameLogic/Session/SessionListener.cs

```diff
@@ -0,0 +1,48 @@
+using Improbable.Gdk.Subscriptions;
+using Improbable.Gdk.Session;
+using UnityEngine;
+
+namespace Fps
+{
+    [WorkerType(WorkerUtils.UnityClient, WorkerUtils.iOSClient, WorkerUtils.AndroidClient)]
+    public class SessionListener : MonoBehaviour
+    {
+        [Require] private SessionReader sessionReader;
+        [Require] private TimerReader timerReader;
+
+        private GameUITimer gameUITimer;
+
+        private void OnEnable()
+        {
+            InitTimer();
+            timerReader.OnCurrentTimeSecondsUpdate += OnCurrentTimeSecondsUpdated;
+        }
+
+        private void OnCurrentTimeSecondsUpdated(uint currentTimeSeconds)
+        {
+            if (gameUITimer == null)
+            {
+                InitTimer();
+            }
+
+            if (gameUITimer != null)
+            {
+                gameUITimer.SynchronizeTime(currentTimeSeconds);
+            }
+        }
+
+        private void InitTimer()
+        {
+            var timerGameObject = GameObject.FindGameObjectWithTag("Timer");
+            if (timerGameObject == null)
+            {
+                return;
```

Comment:
Is this allowed to fail? If not, this should error

Reply:
yeah, it's allowed to fail as it won't break the game if we can't update the timer. You simply won't see the timer